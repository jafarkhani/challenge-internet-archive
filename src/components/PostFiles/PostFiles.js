import React from 'react';
import './PostFiles.scss';

function PostFiles(props){

    const exceptionKeys = ["Unknown", "Metadata"];

    var fileTypes = [];
    props.files.filter(file => exceptionKeys.indexOf(file.format) === -1 )
        .map(file => {
            
            if(fileTypes[file.format] === undefined){
                fileTypes[file.format] = {
                    format : file.format,
                    fileCount : 0,
                    totalSize : 0,
                    name : file.name
                };
            }
            fileTypes[file.format].fileCount++;
            fileTypes[file.format].totalSize += file.size === undefined ? 0 : Math.round(file.size/1024);
        });
    
    let fileRows = Object.keys(fileTypes).map(format => {
        return (
            <li className="list-group-item d-flex justify-content-between align-items-center">
                <a target="blank" href={
                    fileTypes[format].fileCount>1 
                        ? props.urls.downloadCompress + props.identifier + "/format=" + format
                        : props.urls.downloadFile + props.identifier + "/" + fileTypes[format].name }>
                
                {format}{fileTypes[format].fileCount > 1 ? 
                "( " + fileTypes[format].fileCount + " files )" : ""}</a>
                <span className="badge post-files-size">
                    {fileTypes[format].totalSize} MB</span>
            </li> );
    });

    return (
        <ul className="list-group">
            <li className="list-group-item list-group-item-secondary">Download Options</li>
            {fileRows}
        </ul>
    );
}
/**/
export default PostFiles;