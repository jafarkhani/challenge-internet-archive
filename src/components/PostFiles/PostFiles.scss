.post-files-size{
  color: gray;
}