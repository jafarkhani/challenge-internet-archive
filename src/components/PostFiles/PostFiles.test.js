import React from "react";
import renderer from 'react-test-renderer';

import PostFiles from "./PostFiles";

describe("PostFiles", ()=> {

    let mockData=[{
        format: "mp3",
        name : "test",
        size : 150
    }];
    it('snapshot Post Files', () => {
        const tree = renderer.create(<PostFiles 
            identifier= "test"
            urls={{
                downloadFile : "df",
                downloadCompress : "dc"
            }}
            files={mockData} />).toJSON();
        expect(tree).toMatchSnapshot();         
    });

})


