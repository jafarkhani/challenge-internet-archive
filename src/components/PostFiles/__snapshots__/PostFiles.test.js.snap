// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFiles render Post Files 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item list-group-item-secondary"
  >
    Download Options
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
  >
    <a
      href="dftest/test"
      target="blank"
    >
      mp3
      
    </a>
    <span
      className="badge post-files-size"
    >
      0
       MB
    </span>
  </li>
</ul>
`;

exports[`PostFiles snapshot Post Files 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item list-group-item-secondary"
  >
    Download Options
  </li>
  <li
    className="list-group-item d-flex justify-content-between align-items-center"
  >
    <a
      href="dftest/test"
      target="blank"
    >
      mp3
      
    </a>
    <span
      className="badge post-files-size"
    >
      0
       MB
    </span>
  </li>
</ul>
`;
